aws/rust-runtime/aws-inlineable/src/s3_express.rs

```diff
@@ -661,3 +661,124 @@ pub(crate) mod identity_provider {
         }
     }
 }
+
+/// Supporting code for S3 Express runtime plugin
+pub(crate) mod runtime_plugin {
+    use aws_smithy_runtime_api::client::runtime_plugin::RuntimePlugin;
+    use aws_smithy_types::config_bag::{FrozenLayer, Layer};
+    use aws_types::os_shim_internal::Env;
+
+    mod env {
+        pub(super) const S3_DISABLE_EXPRESS_SESSION_AUTH: &str =
+            "AWS_S3_DISABLE_EXPRESS_SESSION_AUTH";
+    }
+
+    #[derive(Debug)]
+    pub(crate) struct S3ExpressRuntimePlugin {
+        config: FrozenLayer,
+    }
+
+    impl S3ExpressRuntimePlugin {
+        pub(crate) fn new(config: FrozenLayer) -> Self {
+            Self::new_with(config, Env::real())
+        }
+
+        fn new_with(config: FrozenLayer, env: Env) -> Self {
+            let mut layer = Layer::new("S3ExpressRuntimePlugin");
+            if config
+                .load::<crate::config::DisableS3ExpressSessionAuth>()
+                .is_none()
+            {
+                match env.get(env::S3_DISABLE_EXPRESS_SESSION_AUTH) {
+                    Ok(value)
+                        if value.eq_ignore_ascii_case("true")
+                            || value.eq_ignore_ascii_case("false") =>
+                    {
```

Comment:
Thanks for catching this. Actually, no. Any other value other than "true" or "false" results in error (doesn't mention case-sensitiveness, though). Other SDKs behave differently, such as for Java, any value that case-insensitively does not equal to "false" is treated as "true". I think we should follow the requirement in that any value that case-insensitively does not equal to "true" or "false" should result in a runtime error.

Reply:
Thanks for caching. Actually no. If a value does not equal to "true" or "false", it should result in an error (although it does not mention case sensitivity). We should follow that and if a value is not case-sensitively equal to "true" or "false", it will be a runtime error (so "TRUE" and "FALSE" are also illegal).

Reply:
Addressed in 3cdd811.